src/PathMappingRepository.php

```diff
@@ -158,23 +159,37 @@ public function hasChildren($path)
     /**
      * {@inheritdoc}
      */
-    protected function addResource($path, FilesystemResource $resource)
+    protected function addFilesystemResource($path, FilesystemResource $resource)
     {
-        // Don't modify resources attached to other repositories
-        if ($resource->isAttached()) {
-            $resource = clone $resource;
-        }
+        $resource->attachTo($this, $path);
```

Comment:
Not easily as `OptimizedPathMappingRepository` does not attach the resource here. Do you think I should override `addResource` in `PathMappingRepository`?